packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts

```diff
@@ -13,16 +15,38 @@ interface Seller {
 }
 
 export interface SKU {
-  itemId: number
+  itemId: string
   sellers: Seller[]
 }
 
-export const useBuyButton = (sku: Maybe<SKU>, quantity: number) => {
+type Options = {
+  isOneClickBuy?: boolean
+}
+
+export const useBuyButton = (
+  sku: Maybe<SKU>,
+  quantity: number,
+  options?: Options
+) => {
   const [loading, setLoading] = useState(false)
   const seller = useBestSeller(sku)
   const orderForm = useOrderForm()
   const disabled = loading || !sku || !orderForm?.value || !seller
 
+  usePixelEvent((e) => {
+    if (options?.isOneClickBuy && e.type !== 'vtex:addToCart') {
+      return
+    }
+
+    const isThisItem = e.data.items[0].id?.toString() === sku?.itemId
+
+    if (!isThisItem) {
+      return
+    }
+
+    window.location.href = '/checkout/'
```

Comment:
we should add this navigation in the next frame so we are sure all events are registered